src/main/java/com/notably/logic/suggestion/SuggestionEngine.java

```diff
@@ -99,34 +100,35 @@ private void suggest(String userInput) {
         final String arguments = matcher.group("arguments");
 
         switch (commandWord) {
-        case OpenSuggestionCommandParser.COMMAND_WORD:
-            return new OpenSuggestionCommandParser(model, pathCorrectionEngine).parse(arguments);
+        case OpenSuggestionArgHandler.COMMAND_WORD:
+            return new OpenSuggestionArgHandler(model, pathCorrectionEngine).handleArg(arguments);
 
-        case DeleteSuggestionCommandParser.COMMAND_WORD:
-            return new DeleteSuggestionCommandParser(model, pathCorrectionEngine).parse(arguments);
+        case DeleteSuggestionArgHandler.COMMAND_WORD:
+            return new DeleteSuggestionArgHandler(model, pathCorrectionEngine).handleArg(arguments);
 
-        case SearchSuggestionCommandParser.COMMAND_WORD:
-            return new SearchSuggestionCommandParser(model).parse(arguments);
+        case SearchSuggestionArgHandler.COMMAND_WORD:
+            return new SearchSuggestionArgHandler(model).handleArg(arguments);
 
-        case NewSuggestionCommandParser.COMMAND_WORD:
-            return new NewSuggestionCommandParser(model).parse(arguments);
+        case NewSuggestionArgHandler.COMMAND_WORD:
+            return new NewSuggestionArgHandler(model).handleArg(arguments);
 
-        case EditSuggestionCommandParser.COMMAND_WORD:
-            return new EditSuggestionCommandParser(model).parse(arguments);
+        case EditSuggestionHandler.COMMAND_WORD:
+            return new EditSuggestionHandler(model).handle();
 
-        case HelpSuggestionCommandParser.COMMAND_WORD:
-            return new HelpSuggestionCommandParser(model).parse(arguments);
+        case HelpSuggestionHandler.COMMAND_WORD:
+            return new HelpSuggestionHandler(model).handle();
 
-        case ExitSuggestionCommandParser.COMMAND_WORD:
-            return new ExitSuggestionCommandParser(model).parse(arguments);
+        case ExitSuggestionHandler.COMMAND_WORD:
+            return new ExitSuggestionHandler(model).handle();
 
         default:
-            return new ErrorSuggestionCommandParser(model).parse(arguments);
+            return new ErrorSuggestionArgHandler(model).handleArg(arguments);
```

Comment:
Actually, this default path will never get executed right? I'm not sure if that's the case. But if it is, would it make sense to not have an `ErrorSuggestionArgHandler` and throw an `AssertionError` here instead?

Reply:
ok!